src/main/java/DataStorageInterface.java

```diff
@@ -0,0 +1,223 @@
+        DataStorage.taskList.add(newTask);
+        return newTask;
+    }
+
+    /**
+     * Adds Event Task to task list.
+     *
+     * @param title User entered title of Task.
+     * @param preposition User entered preposition for date and time.
+     * @param date User entered due date of Task.
+     * @param time User entered timing of Task.
+     * @return Task that is added.
+     * @throws WrongUsageException If event command is used wrongly.
+     */
+    public static Task addEvent(String title, String preposition, LocalDate date, LocalTime time)
+            throws WrongUsageException {
+        Task newTask = new Event(title,preposition,date, time);
+        DataStorage.taskList.add(newTask);
+        return newTask;
+    }
+
+    /**
+     * Removes all tasks from the Task List.
+     */
+    public static void removeAllTasks() {
+        DataStorage.taskList.clear();
+    }
+
+    /**
+     * Adds Tasks from an array to the current Task List.
+     *
+     * @param taskArrayList Tasks to be Added in current Task List.
+     */
+    public static void loadNewTasks(ArrayList<Task> taskArrayList) {
+        DataStorage.taskList.addAll(taskArrayList);
+    }
+
+    /**
+     * Returns String representation of all tasks in the task list.
+     *
+     * @return String representation of all tasks in the task list.
+     */
+    public static String listOfTasks() {
+        StringBuilder acc = new StringBuilder();
+        int i = 0;
+        for (Task t: DataStorage.taskList) {
+            i++;
+            acc.append(String.format("%d. %s\n", i, t));
+        }
+        return acc.toString().isEmpty() ? "There are no Tasks in the list" : acc.toString();
+    }
+
+    /**
+     * Returns String representation of tasks to be saved in an external file.
+     *
+     * @return String representation of tasks to be saved in an external file.
+     */
+    public static String getSaveRepresentation() {
+        StringBuilder acc = new StringBuilder();
+        int i = 0;
+        for (Task t: DataStorage.taskList) {
+            i++;
+            acc.append(String.format("%d. %s\n", i, t.saveStringRepresentation()));
+        }
+        return acc.toString().isEmpty() ? "" : acc.toString();
+    }
+
+    /**
+     * Returns Usage of a single Command.
+     *
+     * @param query Command queried by User.
+     * @return String representation of Command queried by User.
+     * @throws UnknownCommandException If User enters wrong Command.
+     */
+    public static String getUsage(String query) throws UnknownCommandException {
+        for (Command c: DataStorage.commandInit) {
+            if (c.getName().equalsIgnoreCase(query)) {
+                return c.getName() + "\nUsage: "  + c.getUsage() +
+                        "\nDescription: " + c.getDescription();
+            }
+        }
+        throw new UnknownCommandException(query);
+    }
+
+    public static String filteredTasks(String searchQuery) {
```

Comment:
Should there be a Javadoc comment for this public method? Unclear what this method does. Consider adding a verb to the method name as well?